Extension/UqlFunction.php

```diff
@@ -59,15 +66,16 @@ public function getName()
     }
 
     /**
-     * Performs the call to the function defined by this FunctionExtension
-     *
      * @param $arguments
+     * @param Context $context
      *
      * @return mixed
      */
-    public function call($arguments)
+    public function call($arguments, Context $context)
     {
-        return call_user_func_array([$this->instance, $this->method], $arguments);
+        array_unshift($arguments, $context);
```

Comment:
this seems weird... why is the context passed in with the arguments and then just stuck onto the front of the arguments directly after?